File: csharp-netstandard/src/Cloud.Governance.Client/Model/ApiUser.cs


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = Cloud.Governance.Client.Client.OpenAPIDateConverter;

namespace Cloud.Governance.Client.Model
{
    /// <summary>
    /// ApiUser model
    /// </summary>
    [DataContract(Name = "ApiUser")]
    public partial class ApiUser : IEquatable<ApiUser>, IValidatableObject
    {
        /// <summary>
        /// Whether the user is an external user.
        /// </summary>
        /// <value>Whether the user is an external user.</value>
        [DataMember(Name = "isExternalUser", EmitDefaultValue = false)]
        public ExternalUserType? IsExternalUser { get; set; }
        /// <summary>
        /// ApiUser type
        /// </summary>
        /// <value>ApiUser type</value>
        [DataMember(Name = "apiUserType", EmitDefaultValue = false)]
        public ApiUserType? ApiUserType { get; set; }

        /// <summary>
        /// Returns false as ApiUserType should not be serialized given that it's read-only.
        /// </summary>
        /// <returns>false (boolean)</returns>
        public bool ShouldSerializeApiUserType()
        {
            return false;
        }
        /// <summary>
        /// Initializes a new instance of the <see cref="ApiUser" /> class.
        /// </summary>
        /// <param name="id">Object ID.</param>
        /// <param name="loginName">Login name.</param>
        /// <param name="isExternalUser">Whether the user is an external user..</param>
        /// <param name="azureUserType">User type in Azure AD.</param>
        /// <param name="displayName">User display name.</param>
        /// <param name="isGroup">Whether an object is a domain group. (default to false).</param>
        /// <param name="email">User e-mail address.</param>
        public ApiUser(string id = default(string), string loginName = default(string), ExternalUserType? isExternalUser = default(ExternalUserType?), string azureUserType = default(string), string displayName = default(string), bool isGroup = false, string email = default(string))
        {
            this.Id = id;
            this.LoginName = loginName;
            this.IsExternalUser = isExternalUser;
            this.AzureUserType = azureUserType;
            this.DisplayName = displayName;
            this.IsGroup = isGroup;
            this.Email = email;
        }

        /// <summary>
        /// Object ID
        /// </summary>
        /// <value>Object ID</value>
        [DataMember(Name = "id", EmitDefaultValue = true)]
        public string Id { get; set; }

        /// <summary>
        /// Login name
        /// </summary>
        /// <value>Login name</value>
        [DataMember(Name = "loginName", EmitDefaultValue = true)]
        public string LoginName { get; set; }

        /// <summary>
        /// User type in Azure AD
        /// </summary>
        /// <value>User type in Azure AD</value>
        [DataMember(Name = "azureUserType", EmitDefaultValue = true)]
        public string AzureUserType { get; set; }

        /// <summary>
        /// User display name
        /// </summary>
        /// <value>User display name</value>
        [DataMember(Name = "displayName", EmitDefaultValue = true)]
        public string DisplayName { get; set; }

        /// <summary>
        /// Whether an object is a domain group.
        /// </summary>
        /// <value>Whether an object is a domain group.</value>
        [DataMember(Name = "isGroup", EmitDefaultValue = false)]
        public bool IsGroup { get; set; }

        /// <summary>
        /// Whether the user is a local system user.
        /// </summary>
        /// <value>Whether the user is a local system user.</value>
        [DataMember(Name = "isLocalUser", EmitDefaultValue = false)]
        public bool IsLocalUser { get; private set; }

        /// <summary>
        /// Returns false as IsLocalUser should not be serialized given that it's read-only.
        /// </summary>
        /// <returns>false (boolean)</returns>
        public bool ShouldSerializeIsLocalUser()
        {
            return false;
        }

        /// <summary>
        /// User e-mail address
        /// </summary>
        /// <value>User e-mail address</value>
        [DataMember(Name = "email", EmitDefaultValue = true)]
        public string Email { get; set; }

        /// <summary>
        /// User job title
        /// </summary>
        /// <value>User job title</value>
        [DataMember(Name = "jobTitle", EmitDefaultValue = true)]
        public string JobTitle { get; private set; }

        /// <summary>
        /// Returns false as JobTitle should not be serialized given that it's read-only.
        /// </summary>
        /// <returns>false (boolean)</returns>
        public bool ShouldSerializeJobTitle()
        {
            return false;
        }

        /// <summary>
        /// Physical delivery office name of the user
        /// </summary>
        /// <value>Physical delivery office name of the user</value>
        [DataMember(Name = "physicalDeliveryOfficeName", EmitDefaultValue = true)]
        public string PhysicalDeliveryOfficeName { get; private set; }

        /// <summary>
        /// Returns false as PhysicalDeliveryOfficeName should not be serialized given that it's read-only.
        /// </summary>
        /// <returns>false (boolean)</returns>
        public bool ShouldSerializePhysicalDeliveryOfficeName()
        {
            return false;
        }

        /// <summary>
        /// Whether the user is valid.
        /// </summary>
        /// <value>Whether the user is valid.</value>
        [DataMember(Name = "isValid", EmitDefaultValue = false)]
        public bool IsValid { get; private set; }

        /// <summary>
        /// Returns false as IsValid should not be serialized given that it's read-only.
        /// </summary>
        /// <returns>false (boolean)</returns>
        public bool ShouldSerializeIsValid()
        {
            return false;
        }

        /// <summary>
        /// Office 365 tenant ID of the user
        /// </summary>
        /// <value>Office 365 tenant ID of the user</value>
        [DataMember(Name = "tenantId", EmitDefaultValue = true)]
        public string TenantId { get; private set; }

        /// <summary>
        /// Returns false as TenantId should not be serialized given that it's read-only.
        /// </summary>
        /// <returns>false (boolean)</returns>
        public bool ShouldSerializeTenantId()
        {
            return false;
        }

        /// <summary>
        /// User additional data
        /// </summary>
        /// <value>User additional data</value>
        [DataMember(Name = "additionalData", EmitDefaultValue = true)]
        public Dictionary<string, Object> AdditionalData { get; private set; }

        /// <summary>
        /// Returns false as AdditionalData should not be serialized given that it's read-only.
        /// </summary>
        /// <returns>false (boolean)</returns>
        public bool ShouldSerializeAdditionalData()
        {
            return false;
        }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class ApiUser {\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  LoginName: ").Append(LoginName).Append("\n");
            sb.Append("  IsExternalUser: ").Append(IsExternalUser).Append("\n");
            sb.Append("  AzureUserType: ").Append(AzureUserType).Append("\n");
            sb.Append("  DisplayName: ").Append(DisplayName).Append("\n");
            sb.Append("  IsGroup: ").Append(IsGroup).Append("\n");
            sb.Append("  IsLocalUser: ").Append(IsLocalUser).Append("\n");
            sb.Append("  Email: ").Append(Email).Append("\n");
            sb.Append("  JobTitle: ").Append(JobTitle).Append("\n");
            sb.Append("  PhysicalDeliveryOfficeName: ").Append(PhysicalDeliveryOfficeName).Append("\n");
            sb.Append("  IsValid: ").Append(IsValid).Append("\n");
            sb.Append("  TenantId: ").Append(TenantId).Append("\n");
            sb.Append("  AdditionalData: ").Append(AdditionalData).Append("\n");
            sb.Append("  ApiUserType: ").Append(ApiUserType).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as ApiUser);
        }

        /// <summary>
        /// Returns true if ApiUser instances are equal
        /// </summary>
        /// <param name="input">Instance of ApiUser to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(ApiUser input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.Id == input.Id ||
                    (this.Id != null &&
                    this.Id.Equals(input.Id))
                ) && 
                (
                    this.LoginName == input.LoginName ||
                    (this.LoginName != null &&
                    this.LoginName.Equals(input.LoginName))
                ) && 
                (
                    this.IsExternalUser == input.IsExternalUser ||
                    this.IsExternalUser.Equals(input.IsExternalUser)
                ) && 
                (
                    this.AzureUserType == input.AzureUserType ||
                    (this.AzureUserType != null &&
                    this.AzureUserType.Equals(input.AzureUserType))
                ) && 
                (
                    this.DisplayName == input.DisplayName ||
                    (this.DisplayName != null &&
                    this.DisplayName.Equals(input.DisplayName))
                ) && 
                (
                    this.IsGroup == input.IsGroup ||
                    this.IsGroup.Equals(input.IsGroup)
                ) && 
                (
                    this.IsLocalUser == input.IsLocalUser ||
                    this.IsLocalUser.Equals(input.IsLocalUser)
                ) && 
                (
                    this.Email == input.Email ||
                    (this.Email != null &&
                    this.Email.Equals(input.Email))
                ) && 
                (
                    this.JobTitle == input.JobTitle ||
                    (this.JobTitle != null &&
                    this.JobTitle.Equals(input.JobTitle))
                ) && 
                (
                    this.PhysicalDeliveryOfficeName == input.PhysicalDeliveryOfficeName ||
                    (this.PhysicalDeliveryOfficeName != null &&
                    this.PhysicalDeliveryOfficeName.Equals(input.PhysicalDeliveryOfficeName))
                ) && 
                (
                    this.IsValid == input.IsValid ||
                    this.IsValid.Equals(input.IsValid)
                ) && 
                (
                    this.TenantId == input.TenantId ||
                    (this.TenantId != null &&
                    this.TenantId.Equals(input.TenantId))
                ) && 
                (
                    this.AdditionalData == input.AdditionalData ||
                    this.AdditionalData != null &&
                    input.AdditionalData != null &&
                    this.AdditionalData.SequenceEqual(input.AdditionalData)
                ) && 
                (
                    this.ApiUserType == input.ApiUserType ||
                    this.ApiUserType.Equals(input.ApiUserType)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.Id != null)
                    hashCode = hashCode * 59 + this.Id.GetHashCode();
                if (this.LoginName != null)
                    hashCode = hashCode * 59 + this.LoginName.GetHashCode();
                hashCode = hashCode * 59 + this.IsExternalUser.GetHashCode();
                if (this.AzureUserType != null)
                    hashCode = hashCode * 59 + this.AzureUserType.GetHashCode();
                if (this.DisplayName != null)
                    hashCode = hashCode * 59 + this.DisplayName.GetHashCode();
                hashCode = hashCode * 59 + this.IsGroup.GetHashCode();
                hashCode = hashCode * 59 + this.IsLocalUser.GetHashCode();
                if (this.Email != null)
                    hashCode = hashCode * 59 + this.Email.GetHashCode();
                if (this.JobTitle != null)
                    hashCode = hashCode * 59 + this.JobTitle.GetHashCode();
                if (this.PhysicalDeliveryOfficeName != null)
                    hashCode = hashCode * 59 + this.PhysicalDeliveryOfficeName.GetHashCode();
                hashCode = hashCode * 59 + this.IsValid.GetHashCode();
                if (this.TenantId != null)
                    hashCode = hashCode * 59 + this.TenantId.GetHashCode();
                if (this.AdditionalData != null)
                    hashCode = hashCode * 59 + this.AdditionalData.GetHashCode();
                hashCode = hashCode * 59 + this.ApiUserType.GetHashCode();
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}

File: typescript/src/models/ApiUser.ts
/* tslint:disable */
/* eslint-disable */
/*
 * Cloud Governance Api
 *
 * Contact: support@avepoint.com
 */

import { exists, mapValues } from '../runtime';
import {
    ApiUserType,
    ApiUserTypeFromJSON,
    ApiUserTypeFromJSONTyped,
    ApiUserTypeToJSON,
    ExternalUserType,
    ExternalUserTypeFromJSON,
    ExternalUserTypeFromJSONTyped,
    ExternalUserTypeToJSON,
} from './';

/**
 * ApiUser model
 * @export
 * @interface ApiUser
 */
export interface ApiUser {
    /**
     * Object ID
     * @type {string}
     * @memberof ApiUser
     */
    id?: string | null;
    /**
     * Login name
     * @type {string}
     * @memberof ApiUser
     */
    loginName?: string | null;
    /**
     * Whether the user is an external user.
     * @type {ExternalUserType}
     * @memberof ApiUser
     */
    isExternalUser?: ExternalUserType;
    /**
     * User type in Azure AD
     * @type {string}
     * @memberof ApiUser
     */
    azureUserType?: string | null;
    /**
     * User display name
     * @type {string}
     * @memberof ApiUser
     */
    displayName?: string | null;
    /**
     * Whether an object is a domain group.
     * @type {boolean}
     * @memberof ApiUser
     */
    isGroup?: boolean;
    /**
     * Whether the user is a local system user.
     * @type {boolean}
     * @memberof ApiUser
     */
    readonly isLocalUser?: boolean;
    /**
     * User e-mail address
     * @type {string}
     * @memberof ApiUser
     */
    email?: string | null;
    /**
     * User job title
     * @type {string}
     * @memberof ApiUser
     */
    readonly jobTitle?: string | null;
    /**
     * Physical delivery office name of the user
     * @type {string}
     * @memberof ApiUser
     */
    readonly physicalDeliveryOfficeName?: string | null;
    /**
     * Whether the user is valid.
     * @type {boolean}
     * @memberof ApiUser
     */
    readonly isValid?: boolean;
    /**
     * Office 365 tenant ID of the user
     * @type {string}
     * @memberof ApiUser
     */
    readonly tenantId?: string | null;
    /**
     * User additional data
     * @type {{ [key: string]: any; }}
     * @memberof ApiUser
     */
    readonly additionalData?: { [key: string]: any; } | null;
    /**
     * ApiUser type
     * @type {ApiUserType}
     * @memberof ApiUser
     */
    readonly apiUserType?: ApiUserType;
}

export function ApiUserFromJSON(json: any): ApiUser {
    return ApiUserFromJSONTyped(json, false);
}

export function ApiUserFromJSONTyped(json: any, ignoreDiscriminator: boolean): ApiUser {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'id': !exists(json, 'id') ? undefined : json['id'],
        'loginName': !exists(json, 'loginName') ? undefined : json['loginName'],
        'isExternalUser': !exists(json, 'isExternalUser') ? undefined : ExternalUserTypeFromJSON(json['isExternalUser']),
        'azureUserType': !exists(json, 'azureUserType') ? undefined : json['azureUserType'],
        'displayName': !exists(json, 'displayName') ? undefined : json['displayName'],
        'isGroup': !exists(json, 'isGroup') ? undefined : json['isGroup'],
        'isLocalUser': !exists(json, 'isLocalUser') ? undefined : json['isLocalUser'],
        'email': !exists(json, 'email') ? undefined : json['email'],
        'jobTitle': !exists(json, 'jobTitle') ? undefined : json['jobTitle'],
        'physicalDeliveryOfficeName': !exists(json, 'physicalDeliveryOfficeName') ? undefined : json['physicalDeliveryOfficeName'],
        'isValid': !exists(json, 'isValid') ? undefined : json['isValid'],
        'tenantId': !exists(json, 'tenantId') ? undefined : json['tenantId'],
        'additionalData': !exists(json, 'additionalData') ? undefined : json['additionalData'],
        'apiUserType': !exists(json, 'apiUserType') ? undefined : ApiUserTypeFromJSON(json['apiUserType']),
    };
}

export function ApiUserToJSON(value?: ApiUser | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'id': value.id,
        'loginName': value.loginName,
        'isExternalUser': ExternalUserTypeToJSON(value.isExternalUser),
        'azureUserType': value.azureUserType,
        'displayName': value.displayName,
        'isGroup': value.isGroup,
        'email': value.email,
    };
}


